pkg/terraform/exec.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package terraform
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"os/exec"
+)
+
+// Exec interface provides a Command interface. By this way, real or fake implementation of functions is determined.
+type Exec interface {
+	CommandContext(ctx context.Context, name string, dir string, env []string, arg ...string) Command
+}
+
+// RealExec struct implements the CommandContext function that calls the function from the exec package.
+type RealExec struct{}
```

Comment:
```suggestion
type StdExec struct{}
```
or
```suggestion
type OSExec struct{}
```
nitpick

Reply:
or `Std`/`OS` if we put them under `exec` package.

Reply:
Seems like they are in `exec` package now but names still contain exec. Repeating the package name in type/function names is [not considered](https://github.com/golang/go/wiki/CodeReviewComments#package-names) very idiomatic, i.e. instead of `exec.OSExec`, we can use `exec.OS` an it'd still provide enough information.